pkg/services/accesscontrol/eval/permission_test.go

```diff
@@ -0,0 +1,128 @@
+package eval
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type permissionTestCase struct {
+	desc        string
+	permission  Evaluator
+	expected    bool
+	permissions map[string]map[string]struct{}
+}
+
+func TestPermission(t *testing.T) {
+	tests := []permissionTestCase{
+		{
+			desc:       "should evaluate to true",
+			expected:   true,
+			permission: Permission("reports:read", "reports:1"),
+			permissions: map[string]map[string]struct{}{
+				"reports:read": {
+					"reports:1": struct{}{},
+				},
+			},
+		},
+		{
+			desc:       "should evaluate to true for empty scope",
+			expected:   true,
+			permission: Permission("reports:read", ScopeNone),
+			permissions: map[string]map[string]struct{}{
+				"reports:read": {
+					"reports:1": struct{}{},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ok, err := test.permission.Evaluate(test.permissions)
+			assert.NoError(t, err)
+			assert.Equal(t, test.expected, ok)
+		})
+	}
+}
+
+type permissionInjectTestCase struct {
+	desc        string
+	eval        Evaluator
+	expected    bool
+	params      map[string]string
+	permissions map[string]map[string]struct{}
+}
+
+func TestPermission_Inject(t *testing.T) {
+	tests := []permissionInjectTestCase{
+		{
+			desc:     "should inject correct param",
+			expected: true,
+			eval:     Permission("reports:read", Combine("reports", Parameter(":reportId"))),
+			params: map[string]string{
+				":id":       "10",
+				":reportId": "1",
+			},
+			permissions: map[string]map[string]struct{}{
+				"reports:read": {
+					"reports:1": struct{}{},
+				},
+			},
+		},
+		{
+			desc:     "should should fail for nil params",
+			expected: false,
+			eval:     Permission("reports:read", Combine("reports", Parameter(":reportId"))),
+			params:   nil,
+			permissions: map[string]map[string]struct{}{
+				"reports:read": {
+					"reports:1": struct{}{},
+				},
+			},
+		},
+		{
```

Comment:
I added this test for fun:
```go
		{
			desc:     "should should fail for too narrow scope",
			expected: false,
			eval:     Permission("reports:read", Combine("reports", ScopeWildcard)),
			params:   nil,
			permissions: map[string]map[string]struct{}{
				"reports:read": {
					"reports:*": struct{}{},
				},
			},
		},
```
It fails, because we don't actually assess the scope wildcard as a wildcard but as the actual '*' character.